class.cpp

```diff
@@ -0,0 +1,83 @@
+class Minefield
+{
+    private:
+    int r,c;                     //r,c - number of rows,columns
+    int num_mines;               //total number of mines
+    bool is_mine[15][15];        //true if mine is at (i,j), false otherwise
+    bool is_swept[15][15];       //true if swept, false otherwise
+    bool is_q_marked[15][15];    //true if q_marked, false otherwise
+    bool is_flagged[15][15];     //true if flagged, false otherwise
+    short number[15][15];        //stores total number of mines in the eight surrounding tiles
+    char state[15][15];          //store info which can be revealed to player
+    bool loss;                   //becomes true when game is lost, false till then
+    int num_swept;               //holds number of swept tiles
+    
+    void set_field(int i_ex,int j_ex)   //mutator
+    {
+        //fill this up
+        //sets both mines and numbers
+    }
+    
+    
+    public:
+    Minefield(int rows, int cols, int num, int i_ex, int j_ex)  //constructor
+    {
+        r=rows;
+        c=cols;
+        num_mines=num;
+        set_field(i_ex,j_ex);
+    }
+    
+    
+    void flag(int i,int j)  //mutator
+    {
+        if(!is_swept[i][j])
+        is_flagged[i][j]=true;
+    }
+    void deflag(int i,int j)  //mutator
+    {
+        if(!is_swept[i][j])
+        is_flagged[i][j]=false;
+    }
+    void q_mark(int i,int j)  //mutator
+    {
+        if(!is_swept[i][j])
+        is_q_marked[i][j]=true;
+    }
+    void de_q_mark(int i,int j)  //mutator
+    {
+        if(!is_swept[i][j])

```

Comment:
if ( !is_swept[i][j] && is_q_marked[i][j] )

Reply:
Adding that condition will not change the working of the code.
It will add an extra access and evaluation in every call, in exchange for bypassing an assignment in a very small minority
of calls. So it is more time-efficient and readable to not use that condition